resharper/resharper-unity/src/Rider/RiderUnitySharedFilesSavingSuppressor.cs

```diff
@@ -0,0 +1,46 @@
+using JetBrains.Annotations;
+using JetBrains.DocumentManagers;
+using JetBrains.DocumentModel;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Host.Features.Documents;
+using JetBrains.ReSharper.Plugins.Unity.ProjectModel;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Rider
+{
+    /// <summary>
+    /// Removes auto sync through disk for shared files, to avoid unity refresh
+    /// </summary>
+    [SolutionComponent]
+    public class RiderUnitySharedFilesSavingSuppressor : IRiderDocumentSavingSuppressor
+    {
+        [NotNull] private readonly DocumentToProjectFileMappingStorage myDocumentToProjectFileMappingStorage;
+        [NotNull] private readonly ILogger myLogger;
+
+        public RiderUnitySharedFilesSavingSuppressor(
+            [NotNull] DocumentToProjectFileMappingStorage documentToProjectFileMappingStorage,
+            [NotNull] ILogger logger)
+        {
+            myDocumentToProjectFileMappingStorage = documentToProjectFileMappingStorage;
+            myLogger = logger;
+        }
+        
+        
+        public bool ShouldSuppress(IDocument document, bool forceSaveOpenDocuments)
+        {
+            if (!forceSaveOpenDocuments) return false;
+            
+            var projectFile = myDocumentToProjectFileMappingStorage.TryGetProjectFile(document);
+            var isUnitySharedProjectFile = projectFile != null 
+                                           && projectFile.IsShared() 
+                                           && projectFile.GetProject().IsUnityProject();
```

Comment:
I suggest to use  JetBrains.ReSharper.Plugins.Unity.ProjectModel.UnitySolutionTracker.IsUnityGeneratedProject